### ISSUE TITLE
[expo-cli] versioning like expo

### ISSUE THREAD

**Issue by ravi-poonia** [expo/eas-cli#173]
Does eas-cli does not pickup versioning defined in app.json? I tried for ios and it's not working. It picks up the values from info.plist. Am i doing something wrong here. there are no docs for maintaining versions with eas, is there another way to do it?

app.json:
```
{
  "name": "TestApp",
  "displayName": "Test App",
  "expo": {
    "version": "1.0.0",
    "ios": {
      "bundleIdentifier": "com.test.app",
      "buildNumber": "5"
    }
  }
}
```

**Comment by byCedric** [comment]
Hi @ravi-poonia! It depends on the type of project you are using.
- For managed projects, we create the native files for you and update these settings using your app manifest.
- For ejected projects, we use whatever you have configured in your native code. That one will be leading.

If you are using the eject workflow, you might want to try `$ expo eject` again after editing your app manifest. It should update the native files with the latest configuration, just like a managed project 😄 

Hope this helps!

**Comment by ravi-poonia** [comment]
@byCedric I am aware that setting up config in app.json file updates the config for the expo build but it's not working with eas build. I am trying to setup a bitbucket pipeline with auto increment build number. I have shared the sample app.json file above.  When I try to create the build the app.json buildNumber is being ignored, the value from info.plist is being picked up. Am I missing something here? As far as I know, changing build number in app.json should be the same as changing in info.plist. 

Pls let me know if I am misunderstanding any part of the documentation. Thx.

**Comment by byCedric** [comment]
Yes, but that's how it works in the managed workflow. Unfortunately, the bare workflow works a bit different at the moment. We are working hard to unify these two workflows, and bring the best of both worlds together; flexibility from bare, with the feeling from managed.

If you have ejected your project, and building in EAS, the native files take precedent over the app manifest. One of our short-term goals is to use Expo tooling to generate the native files on the go. That's why I asked to run `$ expo eject` (even though you already ejected) if you change properties like the version in your app manifest. Please try that out 😄

**Comment by ravi-poonia** [comment]
I used `react-native init` for the project setup :sweat_smile:
I just wanted a convenient way to auto increment build number during my pipeline execution. I was hoping expo config could help me with that. But it seems that's not implemented yet.

In the end I tried a bunch of things:
•  fastlane increment plugin (didn't work as pipeline was being executed in alpine container)
• iOS build phase to set build number from git commit counts (didn't work, it seems git files are not uploaded with code archive during eas build) 
• iOS build phase to set build version and build number based to current date and time (it worked but seemed kinda patchy and messy so finally rejected)

after many failed attempts, I just created a ruby script to parse the  Info.plist file, increment the build number and update it back to Info.plist file. 

For future adventurers : [gist](https://gist.github.com/ravi-poonia/baaa0b178f07d6558bf9f0462c6d70e1)

I think the script can be simplified some more, but I'm not very experienced with ruby. So, any help with that will be appreciated. Thx
